lib/usd/ui/layerEditor/mayaSessionState.cpp

```diff
@@ -197,19 +197,24 @@ void MayaSessionState::nodeRenamedCB(MObject& obj, const MString& oldName, void*
 {
     if (oldName.length() != 0) {
         auto THIS = static_cast<MayaSessionState*>(clientData);
-        // this does not work:
-        //        if OpenMaya.MFnDependencyNode(obj).typeName == PROXY_NODE_TYPE
-        if (obj.hasFn(MFn::kShape)) {
-            MDagPath dagPath;
-            MFnDagNode(obj).getPath(dagPath);
-            auto shapePath = dagPath.fullPathName();
-
-            StageEntry entry;
-            if (getStageEntry(&entry, shapePath)) {
-                QTimer::singleShot(0, [THIS, entry]() {
-                    Q_EMIT THIS->stageRenamedSignal(entry._displayName, entry._stage);
-                });
-            }
+
+        // doing it on idle give time to the Load Stage to set a file name
+        QTimer::singleShot(0, [THIS, obj]() { THIS->nodeRenamedCBOnIdle(obj); });
+    }
+}
+
+void MayaSessionState::nodeRenamedCBOnIdle(const MObject& obj)
+{
+    // this does not work:
+    //        if OpenMaya.MFnDependencyNode(obj).typeName == PROXY_NODE_TYPE
+    if (obj.hasFn(MFn::kShape)) {
+        MDagPath dagPath;
+        MFnDagNode(obj).getPath(dagPath);
+        auto shapePath = dagPath.fullPathName();
+
+        StageEntry entry;
+        if (getStageEntry(&entry, shapePath)) {
+            Q_EMIT stageRenamedSignal(entry._displayName, entry._stage);
```

Comment:
This "node renamed" callback can be called during file IO but did not have an "Idle" implementation like other callbacks here.  Now nodeNamedCB just uses a QTimer to run nodeRenamedCBOnIdle at a later time.